module/modmanager/manager.go

```diff
@@ -1119,15 +1121,55 @@ func (mgr *Manager) FirstRun(ctx context.Context, conf config.Module) error {
 	for key, val := range moduleEnvironment {
 		cmd.Env = append(cmd.Env, key+"="+val)
 	}
-	cmdOut, err := cmd.CombinedOutput()
 
-	resultLogger := logger.With("path", firstRunPath, "output", string(cmdOut))
+	stdOut, err := cmd.StdoutPipe()
 	if err != nil {
-		resultLogger.Errorw("command failed", "error", err)
 		return err
 	}
+	stdErr, err := cmd.StderrPipe()
+	if err != nil {
+		return err
+	}
+
+	var combined []string
+	var mu sync.Mutex
 
-	resultLogger.Infow("command succeeded")
+	scanOut := bufio.NewScanner(stdOut)
+	go func() {
+		for scanOut.Scan() {
+			out := scanOut.Text()
+			logger.Infow("first run stdio", "output", out)
+			mu.Lock()
+			combined = append(combined, out)
+			mu.Unlock()
+		}
+	}()
+	scanErr := bufio.NewScanner(stdErr)
+	go func() {
+		for scanErr.Scan() {
+			out := scanErr.Text()
+			logger.Warnw("first run stderr", "output", out)
+			mu.Lock()
+			combined = append(combined, out)
+			mu.Unlock()
+		}
+	}()
+	if err := cmd.Start(); err != nil {
+		logger.Errorw("first run failed to start", "error", err)
+		return err
+	}
+	if err := cmd.Wait(); err != nil {
+		logger.Errorw("first run failed", "error", err, "combined output", combined)
+		return err
+	}
+	logger.Infow("first run succeeded", "combined output", combined)
```

Comment:
Because you attached a unique identifier to each process run, I don't think this is necessary. Incidentally, this log can also be problematic. This would get put into the database as a single document and be subject to the 16MB limit.

Reply:
good point, removing